exercises/practice/run-length-encoding/.meta/zcl_rle.clas.abap

```diff
@@ -0,0 +1,62 @@
+CLASS zcl_rle DEFINITION
+  PUBLIC
+  FINAL
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+
+    METHODS encode IMPORTING input         TYPE string
+                   RETURNING VALUE(result) TYPE string.
+
+    METHODS decode IMPORTING input         TYPE string
+                   RETURNING VALUE(result) TYPE string.
+
+ENDCLASS.
+
+
+CLASS zcl_rle IMPLEMENTATION.
+
+  METHOD encode.
+
+    DATA: char       TYPE string,
+          duplicates TYPE string.
+
+    DATA(str) = input.
+
+    WHILE str IS NOT INITIAL.
+
+      FIND REGEX '([\w|\s])(\1*)(.*)' IN str SUBMATCHES char duplicates str  ##SUBRC_OK.
```

Comment:
what does `\1` do? asking because I'm not sure it works with the transpiler, so might be a transpiler bug there

Reply:
Backreference to first capture group (`\2` would be second one etc.). Not sure on which versions it runs, but works on my 7.52 dev system and in DEMO_REGEX_TOY
https://help.sap.com/doc/abapdocu_latest_index_htm/latest/en-US/index.htm?file=abenregex_pcre_syntax_specials.htm

Reply:
hmm, still not sure what it does :) anyhow, lets just merge it, we can work from here

Reply:
Alos in 7.4 
https://help.sap.com/doc/abapdocu_740_index_htm/7.40/en-US/abenregex_syntax_operators.htm
So search for `(word char or space)(same again 0 or more times)(the rest)`

Reply:
It's kind of an 'inline $1'